master/pkg/searcher/grid.go

```diff
@@ -13,6 +13,8 @@ type (
 	// gridSearchState stores the state for grid. Though grid is stateless, it is useful
 	// on restart to know the type of searcher during restore, in the event it must be shimmed.
```

Comment:
blocking: wouldn't adding `PendingTrials` and `RemainingTrials` make this stateful? Is the reasoning the same as what the comment describes for `SearchMethodType`? Also, update the comment if so.